pkg/deploy/cmd/delete.go

```diff
@@ -31,34 +29,11 @@ type DeploymentConfigReaper struct {
 	pollInterval, timeout time.Duration
 }
 
-type updateConfigFunc func(d *deployapi.DeploymentConfig)
-
-// updateConfigWithRetries will try to update a deployment config and ignore any update conflicts.
-func updateConfigWithRetries(dn appsinternal.DeploymentConfigsGetter, namespace, name string, applyUpdate updateConfigFunc) (*deployapi.DeploymentConfig, error) {
-	var config *deployapi.DeploymentConfig
-
-	resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		var err error
-		config, err = dn.DeploymentConfigs(namespace).Get(name, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-		// Apply the update, then attempt to push it to the apiserver.
-		applyUpdate(config)
-		config, err = dn.DeploymentConfigs(namespace).Update(config)
-		return err
-	})
-	return config, resultErr
-}
-
 // pause marks the deployment configuration as paused to avoid triggering new
 // deployments.
 func (reaper *DeploymentConfigReaper) pause(namespace, name string) (*deployapi.DeploymentConfig, error) {
-	return updateConfigWithRetries(reaper.appsClient.Apps(), namespace, name, func(d *deployapi.DeploymentConfig) {
-		d.Spec.RevisionHistoryLimit = kutil.Int32Ptr(0)
-		d.Spec.Replicas = 0
-		d.Spec.Paused = true
-	})
+	patchBytes := []byte(`{"spec":{"paused":true,"replicas":0,"revisionHistoryLimit":0}}`)
```

Comment:
Don't do that manually, you should use `strategicpatch.CreateTwoWayMergePatch` instead.

Reply:
i just got rid of that to avoid extra get to avoid extra latency... since this likely never change, I think this is safe for now